indico/modules/users/client/js/UserDelete.jsx

```diff
@@ -0,0 +1,204 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2024 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import userDeleteURL from 'indico-url:users.user_delete';
+
+import PropTypes from 'prop-types';
+import React, {useEffect, useState} from 'react';
+import ReactDOM from 'react-dom';
+import {Button, Message, Modal, Icon, List, Popup} from 'semantic-ui-react';
+
+import {Translate, Param} from 'indico/react/i18n';
+import {indicoAxios, handleAxiosError} from 'indico/utils/axios';
+
+function UserDeleteDialogBody({firstName, lastName, disabled, inProgress, onDelete, onClose}) {
+  const [countdown, setCountdown] = useState(10);
+  const [isButtonDisabled, setButtonDisabled] = useState(true);
+
+  useEffect(() => {
+    setCountdown(10);
+    setButtonDisabled(true);
+    const timer = setInterval(() => {
+      setCountdown(prevCountdown => {
+        if (prevCountdown <= 1) {
+          clearInterval(timer);
+          setButtonDisabled(false);
+          return 0;
+        }
+        return prevCountdown - 1;
+      });
+    }, 1000);
+    return () => clearInterval(timer);
+  }, []);
+
+  return (
+    <>
+      <Modal.Header>
+        <Translate>
+          Delete <Param name="first_name" value={firstName} />{' '}
+          <Param name="last_name" value={lastName} />?
+        </Translate>
+      </Modal.Header>
+      <Modal.Content>
+        <Message negative icon>
+          <Icon name="warning sign" />
+          <Message.Content>
+            <Message.Header>
+              <Translate>This action is irreversible</Translate>
+            </Message.Header>
+            <Translate>Deleted user accounts cannot be restored.</Translate>
+          </Message.Content>
+        </Message>
+        <Translate as="p">Once deleted, the following will happen:</Translate>
+        <List style={{marginTop: 0}}>
+          <List.Item>
+            <List.Icon name="minus circle" />
+            <List.Content>
+              <Translate>
+                <Param name="first_name" value={firstName} /> will no longer be able to access
+                Indico.
+              </Translate>
+            </List.Content>
+          </List.Item>
+          <List.Item>
+            <List.Icon name="times circle outline" />
+            <List.Content>
+              <Translate>
+                <Param name="first_name" value={firstName} /> will be removed from all events,
+                contributions, and other areas of Indico.
```

Comment:
Should we clarify here that this will NOT remove "persons" from events, ie the user will still show up as a speaker or whatever?

Reply:
Maybe something like "Where possible, {name} will be removed from all events, contributions, and other areas of Indico."?

Reply:
Not sure if that's really clear.. it's not a matter of whether it's possible (it would certainly be possible) but rather that it's not supposed to remove people from those roles since deleting an Indico user does not imply pretending them not having having been e.g. a speaker at some event.

Reply:
I agree it'll be good to clarify this. The the text currently states that it'll be removed, but for example with registrations, the registration is no longer linked to the deleted user but the registration data will continue to exist.

Reply:
What about something like:

{name}'s account will be unliked from all areas of indico, such as events, roles, contributions, e.t.c. Those data/activities will still remain without any further association to {name}

Reply:
I think this is a bit more concise:

"{name} will be removed from all areas of Indico - this does not affect their presence in events as a speaker or other role; they will still be listed in such capacities where applicable."